spec/components/result-view/result-view-spec.js

```diff
@@ -0,0 +1,72 @@
+"use babel";
+
+import React from "react";
+import { shallow } from "enzyme";
+
+import ResultViewComponent from "../../../lib/components/result-view/result-view";
+
+describe("ResultViewComponent", () => {
+  it("should render", () => {
+    const mockStore = {
+      index: 0,
+      outputs: [{}, {}],
+      status: "",
+      executionCount: 0,
+      position: {
+        lineHeight: 0,
+        lineLength: 0,
+        editorWidth: 0,
+        charWidth: 0
+      }
+    };
+
+    const mockKernel = {};
+
+    const destroy = () => {};
+
+    const component = shallow(
+      <ResultViewComponent
+        store={mockStore}
+        kernel={mockKernel}
+        destroy={destroy}
+        showResult={true}
+      />
+    );
+
+    expect(component.type()).not.toBeNull();
+  });
+
+  it("should destroy itself when x is clicked.", () => {
+    const mockStore = {
+      index: 0,
+      outputs: [{}, {}],
+      status: "",
+      executionCount: 0,
+      position: {
+        lineHeight: 0,
+        lineLength: 0,
+        editorWidth: 0,
+        charWidth: 0
+      }
+    };
+
+    const mockKernel = {};
+
+    let destroyInvoked = false;
+    const destroy = () => {
+      destroyInvoked = true;
```

Comment:
This could be simplified using [jasmine spies](https://jasmine.github.io/2.0/introduction.html):

```js
const destroy = jasmine.createSpy("destroy");
const component = shallow(
  <ResultViewComponent
    store={mockStore}
    kernel={{}}
    destroy={destroy}
    showResult={true}
  />
);

component.find(".icon-x").simulate("click");
expect(destroy).toHaveBeenCalled();
```

Reply:
@hpierce1102 did you want to take a crack at using Jasmine spies here?